netty/src/main/java/io/grpc/netty/Utils.java

```diff
@@ -132,16 +134,18 @@ private static final class ByteBufAllocatorPreferHeapHolder {
     }
   }
 
+  private static final MessageFormat BYTEBUF_ALLOCATOR_MESSAGE = new MessageFormat(
+        "Using custom allocator: forceHeapBuffer={0}, defaultPreferDirect={1}", Locale.ROOT);
+
   public static ByteBufAllocator getByteBufAllocator(boolean forceHeapBuffer) {
     if (Boolean.parseBoolean(
             System.getProperty("io.grpc.netty.useCustomAllocator", "true"))) {
       boolean defaultPreferDirect = PooledByteBufAllocator.defaultPreferDirect();
       logger.log(
           Level.FINE,
-          String.format(
-              "Using custom allocator: forceHeapBuffer=%s, defaultPreferDirect=%s",
-              forceHeapBuffer,
-              defaultPreferDirect));
+          BYTEBUF_ALLOCATOR_MESSAGE.format(
```

Comment:
This would still format the string eagerly. I was suggesting to let the logger do the formatting if the level is enabled:
```java
logger.log(
    Level.FINE,
    "Using custom allocator: forceHeapBuffer={0}, defaultPreferDirect={1}",
    new Object[] {forceHeapBuffer, defaultPreferDirect});
```

Reply:
Yeah sorry. Can't manage to import the project on IntelliJ so doing weird things...